### DISCUSSION TITLE
Auto-document config values available for all diagrams

### DISCUSSION THREAD

**Discussion by sneakers-the-rat** [mermaid-js/mermaid#5135] [General]
Didn't think this deserved an issue but thought it was worth raising:

Initially I was going to post "Document ability for ER diagram to use layoutDirection" but it seems like this is a more general need across the package.

Motivating example - ER diagrams of a hierarchical system can get very very tall. Sometimes that doesn't fit with inline examples, eg. between paragraphs where you might want to layout a few entities horizontally to show their relationship.

The ER plot has a pretty well documented config object here, where `layoutDirection` is supported: https://github.com/mermaid-js/mermaid/blob/d8e26dcb17fc31e7d901c95920d003578026466d/packages/mermaid/src/config.type.ts#L915

but that's not shown anywhere in the docs. Instead only the `fill` and `stroke` fields are documented. I don't have time at the moment to look through how the documentation is generated, but it seems like it should be straightforward to autogenerate a table from the config object itself so the docs don't go out of sync with the library.

Doing all the work to implement features and then have nobody use them because they're not in the docs is such a waste! Let's honor how much work y'all have put in and make it visible, ya???

**Comment by roke-julian-lockhart** [comment]
> https://github.com/orgs/mermaid-js/discussions/5135#discussion-5951285

@sneakers-the-rat, I don't see why this isn't better as an issue.

**Reply by sneakers-the-rat** [reply]
If devs want it to be an issue they can raise an issue off of it  🤷‍♀️ be my guest

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/mermaid-js/discussions/5135#discussioncomment-8586284

@sneakers-the-rat, I've submitted https://github.com/mermaid-js/mermaid/issues/5542#issue-2315721544.
